osbuild/buildroot.py

```diff
@@ -140,10 +140,7 @@ def __enter__(self):
             os.makedirs(self.var, exist_ok=True)
             # Ensure /var/tmp is available, see
             #  https://github.com/osbuild/bootc-image-builder/issues/223
-            try:
-                os.symlink("/tmp", os.path.join(self.var, "tmp"))
-            except FileExistsError:
-                pass
+            os.makedirs(os.path.join(self.var, "tmp"), 0o1777, exist_ok=True)
```

Comment:
OK, yeah.  But hmm...at least in the `bootc install` flow we go out of our way to mount the host's `/var/tmp`. 

And systemd with `PrivateTmp` actually sets up a private *subdirectory* of `/var/tmp` which is arguably the best fix here.  If we just point `/var/tmp` at the container overlayfs that's ok, but I would argue to have it point to something underneath one of our volumes (because this just works better in general than having it on the transient overlay).

Reply:
The reason I'm mentioning this is that some of the things that sophisticated build tools often want to do is exactly things like set `security.selinux` or other things that aren't supported on overlayfs (or I guess really overlayfs-mounted-with-context as configured by podman).

Our use cases are already privileged, so having the bwrap invocation not have additional limitations on what it can write to `/var/tmp` would seem to me to be a good idea.

Reply:
`self.var` in the buildroot is a directory under the osbuild store (see https://github.com/osbuild/osbuild/blob/ba08a524a4545a57cd6b9a1ec295e4741262ecea/osbuild/buildroot.py#L91, https://github.com/osbuild/osbuild/blob/ba08a524a4545a57cd6b9a1ec295e4741262ecea/osbuild/pipeline.py#L148 and https://github.com/osbuild/osbuild/blob/ba08a524a4545a57cd6b9a1ec295e4741262ecea/osbuild/objectstore.py#L324).

This is necessary because osbuild expects that its store isn't split into multiple partitions, because its performance heavily relies on atomic moves and CoW (where available). We also don't expect people to put an osbuild store on overlayfs. We should document both these premises in osbuild.

bootc-image-builder "ensures" that osbuild's store is on a "real disk" by specifying a VOLUME for it: https://github.com/osbuild/bootc-image-builder/blob/e21a5fa3303187f63eb25cc8e735ef7996622451/Containerfile#L26

Yes, this probably have some corner cases, but this is something we should address in bib if necessary.